lms/djangoapps/lti_provider/views.py

```diff
@@ -97,7 +97,7 @@ def lti_launch(request, course_id, usage_id):
     for key in lti_keys:
         if key in params:
             lti_params[lti_keys[key]] = params[key]
-    authenticate_lti_user(request, params['user_id'], lti_consumer, lti_params)
+    authenticate_lti_user(request, params['user_id'], params['roles'], lti_consumer, lti_params)
```

Comment:
Agree, no role or an unrecognized role gets you a normal (student) user.

I think @dmitry-Viskov's concern was that if the `roles` key is not present in params, it'll blow up. You can use `params.get('roles')` instead, which will return `None` if the key is not found.